src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -10,6 +10,11 @@
 use Illuminate\Support\Str;
 use LogicException;
 
+/**
+ * @template TKey of array-key
+ * @template T
```

Comment:
Can we avoid using templates in the core?

Reply:
`@template T` is exactly how PhpStorm does it in their example here: https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-release/#collections_with_template